packages/player/src/components/video.tsx

```diff
@@ -51,35 +59,48 @@ export const Video: React.FC<VideoProps> = ({
   //to the videoService and update our state machine context
   const videoElemRef = React.useRef<any>(null)
   const videoService = useVideo()
+  const paused = useSelector(videoService, selectPaused)
+  const readyState = useSelector(videoService, selectReadyState)
+  const currentTime = useSelector(videoService, selectCurrentTime)
+  const autoplay = useSelector(videoService, selectAutoplay)
 
   useMetadataTrackList()
 
+  React.useEffect(() => {
+    videoService.send({type: 'REGISTER', videoRef: videoElemRef})
+
+    // sometimes the event handlers aren't registered before the
+    // `canPlay` event is fired (caching, for instance) so we want
+    // to check and see if it's ready as soon as we get a ref to
+    // the HTMLVideoElement and "manually" fire an event
+
+    videoElemRef.current.readyState > 3 && videoService.send('LOADED')
+  }, [videoElemRef])
```

Comment:
I think this will need to update on changes to `videoElemRef.current` versus the ref itself

It's additionally problematic in that `current` can be null/undefined so this would cause a runtime error